examples/cwms_versioned_ts_examples.py

```diff
@@ -0,0 +1,259 @@
+        ''')
+    print(f"Storing unversioned ts {unversioned_ts['name']}")
+    ts_api.write_ts(data=unversioned_ts, timezone="UTC")
+
+    timezone = pytz.timezone("UTC")
+    begin = timezone.localize(datetime(2008, 5, 1, 15, 0, 0))
+    end = timezone.localize(datetime(2008, 5, 1, 18, 0, 0))
+
+    unversioned_ts_dict = ts_api.retrieve_ts_json(
+        tsId=unversioned_ts['name'], office_id="SWT", begin=begin, end=end)
+    print(unversioned_ts_dict)
+
+def run_max_agg_ts_examples():
+    print("------Running through versioned ts examples-------")
+
+    location = '''
+            {
+              "name": "TEST",
+              "latitude": 0,
+              "longitude": 0,
+              "active": true,
+              "public-name": "CWMS TESTING",
+              "long-name": "CWMS TESTING",
+              "description": "CWMS TESTING",
+              "timezone-name": "America/Los_Angeles",
+              "location-kind": "PROJECT",
+              "nation": "US",
+              "state-initial": "CA",
+              "county-name": "Yolo",
+              "nearest-city": "Davis, CA",
+              "horizontal-datum": "NAD83",
+              "vertical-datum": "NGVD29",
+              "elevation": 320.04,
+              "bounding-office-id": "SPK",
+              "office-id": "SPK"
+            }
+            '''
+    headers = {
+        "Content-Type": constants.HEADER_JSON_V2
+    }
+    print("Storing location TEST")
+    ts_api.get_session().post("locations", params=None,
+                              headers=headers, data=location)
+
+    versioned_ts = json.loads(
+        '''
+        {
+          "office-id": "SWT",
+          "name": "TEST.Flow.Inst.1Hour.0.MockTestMaxAgg",
+          "units": "CFS",
+          "version-date": "2021-06-20T08:00:00-0000[UTC]",
+          "values": [
+            [
+              1209654000000,
+              4,
+              0
+            ],
+            [
+              1209657600000,
+              3,
+              0
+            ],
+            [
+              1209661200000,
+              2,
+              0
+            ]
+          ]
+        }
+        ''')
+    print(f"Storing versioned ts 1 {versioned_ts['name']}")
+    ts_api.write_ts(data=versioned_ts, timezone="UTC")
+
+    versioned_ts2 = json.loads(
+        '''
+        {
+          "office-id": "SWT",
+          "name": "TEST.Flow.Inst.1Hour.0.MockTestMaxAgg",
+          "units": "CFS",
+          "version-date": "2021-06-21T08:00:00-0000[UTC]",
+          "values": [
+            [
+              1209664800000,
+              1,
+              0
+            ]
+          ]
+        }
+        ''')
+
+    print(f"Storing versioned ts 2 {versioned_ts2['name']}")
+    ts_api.write_ts(data=versioned_ts2, timezone="UTC")
+
+    timezone = pytz.timezone("UTC")
+    begin = timezone.localize(datetime(2008, 5, 1, 15, 0, 0))
+    end = timezone.localize(datetime(2008, 5, 1, 18, 0, 0))
+
+    max_agg_ts_dict = ts_api.retrieve_ts_json(
```

Comment:
Curious about thoughts on the timezone. I generally prefer a required timezoned datetime and not parameterizing in the timezone. The implementation here assumes that the datetime does not have a time zone (and stringifies it) and passes in the timezone parameter separately.

In this case, the example is a little confusing in that you are not parameterizing in the timezone, which winds up defaulting to "UTC" on the server. You could have removed the `timezone.localize` and achieved the same effect. 

My recommendation would be to parameterize in the time zone (or we open up a ticket to remove the timezone parameter).